src/modules/currency/wallet/currency-wallet-callbacks.js

```diff
@@ -40,6 +40,64 @@ export function forEachListener (
   }
 }
 
+type ThrottleDetails = {
+  [walletId: string]: {
+    delayCallback: boolean,
+    lastCallbackTime: number,
+    txArray: Array<EdgeTransaction>
+  }
+}
+
+const throttleDetailsNewTx: ThrottleDetails = {}
+const throttleDetailsChangedTx: ThrottleDetails = {}
+
+let throttleRateLimitMs = 5000
+
+function throttledTxCallback (
+  throttleDetails: ThrottleDetails,
+  callback: Function,
+  walletId: string,
+  txArray: Array<EdgeTransaction>
+) {
+  // If this is a unit test, lower throttling to 250ms
+  if (txArray[0].txid.length < 3) {
+    throttleRateLimitMs = 250
+  }
+  if (!throttleDetails[walletId]) {
+    throttleDetails[walletId] = {
+      delayCallback: false,
+      lastCallbackTime: 0,
+      txArray: []
+    }
+  }
+  if (!throttleDetails[walletId].delayCallback) {
+    const now = Date.now()
+    if (
+      now - throttleDetails[walletId].lastCallbackTime >
+      throttleRateLimitMs
+    ) {
+      callback(walletId, txArray)
+      throttleDetails[walletId].txArray = []
+      throttleDetails[walletId].lastCallbackTime = Date.now()
+    } else {
+      console.log(
+        'throttledTxCallback setTimeout delay callback id:' + walletId
+      )
+      throttleDetails[walletId].delayCallback = true
+      throttleDetails[walletId].txArray = txArray
+      setTimeout(() => {
+        callback(walletId, throttleDetails[walletId].txArray)
+        throttleDetails[walletId].delayCallback = false
+        throttleDetails[walletId].txArray = []
+        throttleDetails[walletId].lastCallbackTime = Date.now()
+      }, throttleRateLimitMs)
+    }
+  } else {
+    Array.prototype.push.apply(throttleDetails[walletId].txArray, txArray)
```

Comment:
Better syntax is `throttleDetails[walletId].txArray.push(...txArray)` - it transpiles to exactly what you typed, but is a lot more understandable.